### DISCUSSION TITLE
Github package registry - Private package not found

### DISCUSSION THREAD

**Discussion by WaqasIbrahim** [community/community#25703] [Packages]
<p>I have a private repository and I was able to publish a private package successfully. But when I try to install the package, It gives a 404 error. I cannot download the package from the package landing page and the assets links to <a href="https://notfound.pkg.github.com/" rel="nofollow noopener">https://notfound.pkg.github.com/</a>.</p>
<p>I changed the visibility to public and the installation and assets’ link worked fine.</p>

**Comment by jonmckee** [comment]
<p>I’m having this exact same problem.</p>

**Comment by mxxk** [comment]
<p>Thanks for creating a dedicated issue for this, <a class="mention" href="/waqasibrahim">@waqasibrahim</a> (the <a href="https://github.community/t/package-not-found-in-the-github-registry/16251">previous issue</a> wasn’t getting any attention since it’s marked fixed for an unrelated reason).</p>
<p>Here are the steps to reproduce:</p>
<ol>
<li>Create a new <em>private</em> repository. Per <a href="https://github.com/npm/npm/issues/7351" rel="nofollow noopener">npm package naming requirements</a>, make sure the repo owner and name (i.e. <span class="mention">@repo-owner</span>/repo-name) are both lowercase.</li>
<li>
<a href="https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line" rel="nofollow noopener">Create a personal access token</a> with the following scopes:</li>
</ol>
<ul>
<li>write:packages</li>
<li>read:packages</li>
<li>repo</li>
</ul>
<ol start="3">
<li>
<p><a href="https://help.github.com/en/github/managing-packages-with-github-packages/configuring-npm-for-use-with-github-packages" rel="nofollow noopener">Authenticate against the GitHub NPM registry</a> using your GitHub username and personal access token:</p>
<p>$ npm login --scope <span class="mention">@repo-owner</span> --registry <a href="https://npm.pkg.github.com" rel="nofollow noopener">https://npm.pkg.github.com</a></p>
</li>
<li>
<p>Use npm init to create a new npm package called <span class="mention">@repo-owner</span>/repo-name.</p>
</li>
<li>
<p>Publish a new (and only) version of the NPM package with npm publish.</p>
</li>
<li>
<p>Check that the package is visible with npm view:</p>
<p>$ npm view <span class="mention">@repo-owner</span>/repo-name<br>
<span class="mention">@repo-owner</span>/repo-name@0.0.1 | ISC | deps: 0 | versions: 1<br>
…</p>
</li>
</ol>
<p>Now, try to install the package:</p>
<pre><code>$ npm install -g @repo-owner/repo-name
npm ERR! code E404
npm ERR! 404 Not Found - GET https://npm.pkg.github.com/download/@repo-owner/repo-name/0.0.1/8d84896fe3f54d8ce56962cf1f7fc760e56efe52f2d471cf703103c476d3c9a9
</code></pre>
<p>Check that though the GitHub UI is showing the new version, its tarball links to <a href="https://notfound.pkg.github.com/" rel="nofollow noopener">https://notfound.pkg.github.com/</a>:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8493055cb89fbdc4ef64294ab0345545e1018d0a.png" data-download-href="/uploads/short-url/iUO13raQnWeOnQAcskdBwOv3awa.png?dl=1" title="github-package-npm.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2456381/181092735-d01b6386-0b25-4c26-9d17-189279a69f31.png" alt="github-package-npm.png" data-base62-sha1="iUO13raQnWeOnQAcskdBwOv3awa" width="690" height="355" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8493055cb89fbdc4ef64294ab0345545e1018d0a_2_690x355.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8493055cb89fbdc4ef64294ab0345545e1018d0a_2_1035x532.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/8/8493055cb89fbdc4ef64294ab0345545e1018d0a_2_1380x710.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8493055cb89fbdc4ef64294ab0345545e1018d0a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">github-package-npm.png</span><span class="informations">1983×1021 182 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Reply by inyeong-kang** [reply]
You're the best!
Thanks a lot.

**Comment by wpitallo** [comment]
<p>I am also getting this.</p>

**Comment by freshly-pressed-trousers** [comment]
<p>Same isssue here, which is a shame as when github packages was released it looked like a perfect solution for consolidating a lot of disparate repository guff I’ve had to deal with.</p>

**Comment by scottwestover** [comment]
<p>I am having the same issue.</p>

**Comment by Alappin** [comment, marked as answer]
<p>I can confirm that the problem has been fixed and might have been related to some “data transfer” billing issues at github. Support eventually cameback to me after 2 weeks and gave me the following response.</p>
<hr>
<p>Hey there *NAME RETRACTED*,</p>
<p>Thanks for your patience here, we believe there may have been some issues with rounding the data.</p>
<p>Can I ask you to:</p>
<ul>
<li>Check, and view your usage over at: <a href="https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/viewing-your-github-packages-usage" rel="nofollow noopener">https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/viewing-your-github-packages-usage</a>
</li>
<li>Try making this installation request again.</li>
</ul>
<p>Usage should be more accurately reflected, and you should be able to start using the package registry once more.</p>
<p>Thanks<br>
*NAME RETRACTED*</p>
<hr>

**Comment by anthonator** [comment]
<p>Is a personal access token still required?</p>

**Comment by anthonator** [comment]
<p>Yes, it is. <code>secrets.GITHUB_TOKEN</code> does not work.</p>

**Comment by anthonator** [comment]
<p>We’ve also been getting charged for data transfer out when using personal access tokens to get a docker image hosted on the GitHub Package Registry from GitHub Actions. This is supposed to be free.</p>

**Comment by cyrus-za** [comment]
<p>I am still experiencing this issue, but only when installing with yarn. Not when installing with npm. So strange!</p>

**Comment by joaogarin** [comment]
<p>Hello all,</p>
<p>I started also receiving today errors for installing packages (private) via circle ci…any ideas if there is anything wrong happenig? I feel like if it was a token problem I would not be getting a 404 but rather a permission denied…locally I can install the packages no problem…</p>
<p>Anyone else experiencing some issues recently on this?</p>
